docs/src/quickstart.md

```diff
@@ -68,37 +68,58 @@ c_d = similar(a_d)
 
 Next, we will define the GPU kernel that does the actual computation:
 
 ```julia
 function vadd!(c, a, b)
-    i = threadIdx().x
+    i = workitemIdx().x
     c[i] = a[i] + b[i]
     return
 end
 ```
 
-This simple kernel starts by getting the current thread ID using [`threadIdx`](@ref) and then performs the addition of the elements from `a` and `b`, storing the result in `c`.
+This simple kernel starts by getting the current thread ID using [`workitemIdx`](@ref) and then performs the addition of the elements from `a` and `b`, storing the result in `c`.
 
 Notice how we explicitly specify that this function does not return a value by adding the `return` statement.
 This is necessary for all GPU kernels and we can enforce it by adding a `return`, `return nothing`, or even `nothing` at the end of the kernel.
-If this statement is omitted Julia will return the value of the last evaluated expression, in this case a `Float64`.
+If this statement is omitted, Julia will attempt to return the value of the last evaluated expression, in this case a `Float64`, which will cause a compilation failure as kernels cannot return values.
 
-The easiest way to launch a GPU kernel is with the [`@roc`](@ref) macro, specifying that we want `N` threads and calling it like an ordinary function:
+The easiest way to launch a GPU kernel is with the [`@roc`](@ref) macro, specifying that we want a single work group with `N` work items and calling it like an ordinary function:
 
 ```julia
-@roc threads=N vadd!(c_d, a_d, b_d)
+@roc groupsize=N vadd!(c_d, a_d, b_d)
 ```
 
-Keep in mind that kernel launches are asynchronous, meaning that you need to do some kind of synchronization before you use the result (this is not necessary in the REPL).
+Keep in mind that kernel launches are asynchronous, meaning that you need to do some kind of synchronization before you use the result.
 For instance, you can call `wait()` on the returned HSA signal value:
 
 ```julia
-wait(@roc threads=N vadd!(c_d, a_d, b_d))
+wait(@roc groupsize=N vadd!(c_d, a_d, b_d))
 ```
 
+!!! warning "Naming conventions"
+    Throughout this example we use terms like "work group" and "work item".
+    These terms are used by the Khronos consortium and their APIs including OpenCL and Vulkan, as well as the HSA foundation.
+
+    NVIDIA, on the other hand, uses some different terms in their CUDA API, which might be confusing to some users porting their kernels from CUDAnative to AMDGPUnative.
+    As a quick summary, here is a mapping of the most common terms:
+    
+    | AMDGPUnative | CUDAnative |
+    |:---:|:---:|
+    | [`workitemIdx`](@ref) | [`threadIdx`](@ref) |
+    | [`workgroupIdx`](@ref) | [`blockIdx`](@ref) |
+    | [`workgroupDim`](@ref) | [`blockDim`](@ref) |
+    | [`gridDim`](@ref) | No equivalent |
+    | [`gridDimWG`](@ref) | `gridDim` |
+    | `groupsize` | `threads` |
+    | `gridsize` | `blocks` |
```

Comment:
This is actually false (as I found out recently, hence the kernel indexing PR I merged the other day), `gridsize == blocks*threads`.

Reply:
Hmm, I thought that `gridsize` is an alias for `blocks` (referring to kernel launch parameters), as defined in `src/execution.jl:14`:
```julia
    alias_kws    = Dict(:blocks=>:gridsize, :threads=>:groupsize,
```

Is that outdated or am I missing something here?

Reply:
Damnit... yeah I screwed that one up too, I'll have to fix that logic as well. Don't worry about the `alias_kws` for this PR, but if you can change `blocks` to `blocks*threads` in the docs, then I think we're good to merge.

Reply:
oh ok, sure!